buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy

```diff
@@ -347,9 +347,16 @@ class ClusterFormationTasks {
 
         Task writeConfig = project.tasks.create(name: name, type: DefaultTask, dependsOn: setup)
         writeConfig.doFirst {
-            String unicastTransportUri = node.config.unicastTransportUri(seedNode, node, project.ant)
-            if (unicastTransportUri != null) {
-                esConfig['discovery.zen.ping.unicast.hosts'] = "\"${unicastTransportUri}\""
+            if (node.nodeVersion.onOrAfter("6.5.0-SNAPSHOT")) {
+                // Don't force discovery provider if one is set by the test cluster specs already
+                if (esConfig.containsKey('discovery.zen.hosts_provider') == false) {
+                    esConfig['discovery.zen.hosts_provider'] = 'file'
+                }
+            } else {
+                String unicastTransportUri = node.config.unicastTransportUri(seedNode, node, project.ant)
+                if (unicastTransportUri != null) {
+                    esConfig['discovery.zen.ping.unicast.hosts'] = "\"${unicastTransportUri}\""
```

Comment:
I think if `unicastTransportUri == null` we should also set `discovery.zen.ping.unicast.hosts: []` or else we will attempt to ping ports `9300-9304` too (this is unrelated to this change, I just noticed it).